packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -388,6 +389,43 @@ class RelativePositionedTransition extends AnimatedWidget {
   }
 }
 
+/// Animated version of a [DecoratedBox] that animates the different properties
+/// of its [Decoration].
+class DecoratedBoxTransition extends AnimatedWidget {
+  /// Creates an animated [DecorationBox] whose [Decoration] animation updates
+  /// the widget. 
+  /// 
+  /// `decoration` and `position` cannot be null.
+  /// 
+  /// Also see [DecoratedBox].
+  DecoratedBoxTransition({
+    Key key,
+    @required this.decoration,
+    this.position: DecorationPosition.background,
+    @required this.child,
+  }) : super(key: key, listenable: decoration);
+
+  /// Animation of the decoration to paint.
+  ///
+  /// Can be created using a [DecorationTween].
```

Comment:
I think we usually also doc a link to BoxDecoration because that's by far the most common kind of Decoration to use and it saves the reader a step.  Something like:

```
Typically a [BoxDecoration].
```

although you might want to adapt that somewhat with the animation in the mix.